modules/configuration/src/main/java/org/apache/ignite/configuration/internal/rest/JsonConverter.java

```diff
@@ -0,0 +1,304 @@
+                JsonObject innerJsonNode = new JsonObject();
+
+                jsonObjectsStack.push(innerJsonNode);
+
+                node.traverseChildren(this);
+
+                jsonObjectsStack.pop();
+
+                addToParent(key, innerJsonNode);
+
+                return innerJsonNode;
+            }
+
+            /** {@inheritDoc} */
+            @Override public <N extends InnerNode> JsonElement visitNamedListNode(String key, NamedListNode<N> node) {
+                JsonObject namedListJsonNode = new JsonObject();
+
+                jsonObjectsStack.push(namedListJsonNode);
+
+                for (String subkey : node.namedListKeys())
+                    node.get(subkey).accept(subkey, this);
+
+                jsonObjectsStack.pop();
+
+                addToParent(key, namedListJsonNode);
+
+                return namedListJsonNode;
+            }
+
+            /** */
+            @NotNull private JsonElement toJsonLeaf(Serializable val) {
+                if (val == null)
+                    return JsonNull.INSTANCE;
+
+                Class<? extends Serializable> valClass = val.getClass();
+
+                if (!valClass.isArray())
+                    return toJsonPrimitive(val);
+
+                JsonArray jsonArray = new JsonArray();
+
+                for (int i = 0; i < Array.getLength(val); i++)
+                    jsonArray.add(toJsonPrimitive(Array.get(val, i)));
+
+                return jsonArray;
+            }
+
+            /** */
+            @NotNull private JsonElement toJsonPrimitive(Object val) {
+                if (val == null)
+                    return JsonNull.INSTANCE;
+
+                if (val instanceof Boolean)
+                    return new JsonPrimitive((Boolean)val);
+
+                if (val instanceof String)
+                    return new JsonPrimitive((String)val);
+
+                if (val instanceof Number)
+                    return new JsonPrimitive((Number)val);
+
+                assert false : val;
+
+                throw new IllegalArgumentException(val.getClass().getCanonicalName());
+            }
+
+            /**
+             * Add subelement to the paretn JSON object if it exists.
+             *
+             * @param key Key for the passed JSON element.
+             * @param jsonElement JSON element to add to the parent.
+             */
+            private void addToParent(String key, JsonElement jsonElement) {
+                if (!jsonObjectsStack.isEmpty())
+                    jsonObjectsStack.peek().add(key, jsonElement);
+            }
+        };
+    }
+
+    /** */
+    public static ConfigurationSource jsonSource(JsonElement jsonElement) {
+        //TODO IGNITE-14372 Finish this implementation.
+        return new JsonObjectConfigurationSource(new ArrayList<>(), jsonElement.getAsJsonObject());
+    }
+
+    private static class JsonObjectConfigurationSource implements ConfigurationSource {
```

Comment:
We have several other implementations of Configuration source.
Does it need to create a new one?